src/sql-parser/src/parser.rs

```diff
@@ -4619,17 +4619,31 @@ impl<'a> Parser<'a> {
     }
 
     /// Parse a query expression, i.e. a `SELECT` statement optionally
-    /// preceeded with some `WITH` CTE declarations and optionally followed
+    /// preceded with some `WITH` CTE declarations and optionally followed
     /// by `ORDER BY`. Unlike some other parse_... methods, this one doesn't
     /// expect the initial keyword to be already consumed
     fn parse_query(&mut self) -> Result<Query<Raw>, ParserError> {
         self.checked_recur_mut(|parser| {
             let cte_block = if parser.parse_keyword(WITH) {
                 if parser.parse_keyword(MUTUALLY) {
                     parser.expect_keyword(RECURSIVE)?;
-                    CteBlock::MutuallyRecursive(
-                        parser.parse_comma_separated(Parser::parse_cte_mut_rec)?,
-                    )
+                    let max_iterations = if parser.parse_keyword(MAXITERATIONS) {
+                        let parsed = parser.parse_literal_uint()?;
+                        if parsed == 0 {
+                            return parser_err!(
+                                parser,
+                                parser.peek_prev_pos(),
+                                "MAXITERATIONS has to be greater than 0"
```

Comment:
This logic should live in the planner, not the parser.

Reply:
Ok, I moved it.

Reply:
The PR should be in it's final form now. Could you please do a final check @mjibson?